servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/publish/jmx/OSGIMBeanPublishBehaviour.java

```diff
@@ -47,17 +47,20 @@ public long getBundleId(MBeanServerConnection mbsc,
 			for (Object row : rows) {
 				if (row instanceof CompositeData) {
 					CompositeData cd = (CompositeData) row;
-					String bsn = cd.get("SymbolicName").toString();
-					String id = cd.get("Identifier").toString();
-					String ver = cd.get("Version").toString();
-					if (version != null) {
-						if (bsn.equals(bundleSymbolicName) && ver.equals(version)) {
-							return Long.parseLong(id);
-						}	
-					} else {
-						// if we don't have a version we take the first best
-						if (bsn.equals(bundleSymbolicName)) {
-							return Long.parseLong(id);
+					Object symbolicName = cd.get("SymbolicName");
+					if(symbolicName != null){
+						String bsn = symbolicName.toString();
+						String id = cd.get("Identifier").toString();
```

Comment:
I didn't encounter the case during development but it can be proper. I will check for it too